src/libs/IOUUtils.js

```diff
@@ -1,31 +1,80 @@
 import _ from 'underscore';
+import Onyx from 'react-native-onyx';
+import lodashGet from 'lodash/get';
 import CONST from '../CONST';
+import ONYXKEYS from '../ONYXKEYS';
+
+let currencyList = {};
+Onyx.connect({
+    key: ONYXKEYS.CURRENCY_LIST,
+    callback: (val) => {
+        if (_.isEmpty(val)) {
+            return;
+        }
+
+        currencyList = val;
+    },
+});
+
+/**
+ * Returns the number of digits after the decimal separator for a specific currency.
+ * For currencies that have decimal places > 2, floor to 2 instead:
+ * https://github.com/Expensify/App/issues/15878#issuecomment-1496291464
+ *
+ * @param {String} currency - IOU currency
+ * @returns {Number}
+ */
+function getCurrencyDecimals(currency = CONST.CURRENCY.USD) {
+    const decimals = lodashGet(currencyList, [currency, 'decimals']);
+    return _.isUndefined(decimals) ? 2 : Math.min(decimals, 2);
+}
+
+/**
+ * Returns the currency's minor unit quantity
+ * e.g. Cent in USD
+ *
+ * @param {String} currency - IOU currency
+ * @returns {Number}
+ */
+function getCurrencyUnit(currency = CONST.CURRENCY.USD) {
+    return 10 ** getCurrencyDecimals(currency);
+}
 
 /**
  * Calculates the amount per user given a list of participants
  * @param {Array} participants - List of logins for the participants in the chat. It should not include the current user's login.
  * @param {Number} total - IOU total amount
+ * @param {String} currency - IOU currency
  * @param {Boolean} isDefaultUser - Whether we are calculating the amount for the current user
+ * @param {Boolean} shouldConvertTo2DigitsFormat - Whether to convert the amount to 2-digits format
  * @returns {Number}
  */
-function calculateAmount(participants, total, isDefaultUser = false) {
+function calculateAmount(participants, total, currency, isDefaultUser = false, shouldConvertTo2DigitsFormat = false) {
```

Comment:
I think the parameter `shouldConvertTo2DigitsFormat` makes the return of this function harder to understand. 

For me, it would be easier if this function always returned the amount in the "cents" that the backend understand, like it was before. This is equivalent to having `shouldConvertTo2DigitsFormat` always `true`, so always return `(finalAmount * 100) / currencyUnit`

Then, when you use it and you need to show the number with decimal places in `this.props.numberFormat(myIOUAmount / currencyUnit`, you just divide by `100` instead of `currencyUnit` like we used to.

Currently `calculateAmount` with `shouldConvertTo2DigitsFormat = false` will return a number that its interpretation is tied to the decimals in the currency, for example:

`IOUUtils.calculateAmount(['a'], 10, 'USD', true, false)` -> `500` (cents)
`IOUUtils.calculateAmount(['a'], 10, 'JPY', true, false)` -> `5` (unit)
`IOUUtils.calculateAmount(['a', 'b'], 10, 'currency with 1 decimal', true, false)` -> `34` (don't know the name, decimal?, but has to be multiplied by 10 to convert to "cents" or divided by 10 to be units)

Reply:
@aimane-chnaif I don't think the unit test should be updated since there weren't tests for `shouldConvertTo2DigitsFormat = true` in the first place (my mistake 😄).

Reply:
`shouldConvertTo2DigitsFormat`'s been removed and tests're updated!